Cesium.CodeGen/Generators/Expressions.cs

```diff
@@ -38,18 +38,38 @@ public static void EmitExpression(FunctionScope scope, Expression expression)
     private static void EmitConstantExpression(FunctionScope scope, ConstantExpression expression)
     {
         var token = expression.Constant;
-        var instruction = token switch
+        var instructions = scope.Method.Body.Instructions;
+        switch (token)
         {
             // TODO: Optimizations like Ldc_I4_0 for selected constants
-            { Kind: CTokenType.IntLiteral } => Instruction.Create(OpCodes.Ldc_I4, int.Parse(token.Text)),
-            { Kind: CTokenType.Identifier, Text: var name } when scope.Variables.TryGetValue(name, out var var) =>
-                Instruction.Create(OpCodes.Ldloc, var),
-            { Kind: CTokenType.Identifier, Text: var name } when scope.Parameters.TryGetValue(name, out var par) =>
-                Instruction.Create(OpCodes.Ldarg, par),
-            _ => throw new Exception($"Constant token not supported: {token.Kind} {token.Text}.")
+            case { Kind: CTokenType.IntLiteral }: 
+                {
+                    var instruction = Instruction.Create(OpCodes.Ldc_I4, int.Parse(token.Text));
+                    instructions.Add(instruction);
+                    break;
+                }
+            case { Kind: CTokenType.CharLiteral }:
+                {
+                    var charValue = UnescapeCharacter(token.Text);
+                    instructions.Add(Instruction.Create(OpCodes.Ldc_I4, (int)charValue));
+                    instructions.Add(Instruction.Create(OpCodes.Conv_U1));
```

Comment:
This is nice catch. Given that you want to have UTF-8 do we need to produce lengths? Or we always assume that C `char` is interpreted on C# side as UTF16 `System.Char`?

Reply:
Ask question here. https://github.com/ForNeVeR/Cesium/issues/57

Reply:
Well, actually you were right and I wasn't.

I think that character literals will only support 7-bit ANSI characters, because everything wider will take more than one byte, and `char` is exactly one byte.